### PR TITLE
Issue/94

### PR DESCRIPTION
**Pull request by hayleyavw**
Draft version of `Contributing to Kordac` page. Please comment with your thoughts, focusing on content rather than formatting (not a lot of effort has gone into formatting just yet).
The Test Suite section has not been written yet, working on it now!

### CODE REVIEW

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+3. Check test suite for errors, and fix any issues that arise, or `log an issue <https://github.com/uccser/cs-field-guide/issues/new>`_.
+4. Detail the changes in ``docs/source/changelog.rst``.
+5. `Complete the release branch <http://nvie.com/posts/a-successful-git-branching-model/#finishing-a-release-branch>`_. Be sure to tag the release with the version number for creating the release on GitHub.
+6. Create the release on `GitHub <https://github.com/uccser/kordac/releases/>`_ on the tagged commit.
+7. Upload a new version of Kordac to PyPI.
+
+.. [1] We follow `Semantic Versioning <http://semver.org/>`_ for our numbering system. The number is used by ``setup.py`` to tell PyPI which version is being uploaded or ``pip`` which version is installed, and also used during the documentation build to number the version of Kordac it was built from.
+
+
+Notes
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Kordac should make use GitHub's features:

- Issue template
- Pull request template
- Contributing page

So GitHub can display these when appropriate.

File: docs/source/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+=======================================
+If you come across a bug in Kordac, please `report it`_ on the repo.
+
+<Bug fix? Add tests.>
+
+.. _report it: https://github.com/uccser/kordac/issues
+
+The Code Base
+=======================================
+If you would like to contribute to Kordac, fork the repository.
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Suggestion:

> If you would like to contribute to Kordac, fork the repository.

to

> If you would like to contribute to Kordac, create a fork of the repository.

File: docs/source/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+- **Tag**
+    This refers to the custom markdown syntax that Kordac processes.
+    
+    For example:
+    
+    .. code-block:: none
+
+      {comment this will be removed by the converter}
+
+      {image file-path='img/totally-real-image.png' alt='process me'}
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Should use double quotes for tag parameters to be consistent with examples.

File: docs/source/contributing.rst [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+- ``KordacResult()``
+    The object returned by ``Kordac()`` containing:
+    	- Converted html string
+    	- Title
+    	- Required files (images, interactives, scratch images, page scripts)
+    	- Heading tree
+    	- Required glossary terms
+
+- ``KordacExtension()``
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Probably want to describe the relationship between `Kordac`, `KordacExtension`, and `markdown` in more detail, possibly with a diagram. It would be confusing to understand why Kordac isn't just an extension. Explain why it's also a markdown wrapper.

File: docs/source/contributing.rst [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    Inherits the ``Extension`` class from Markdown.
+    This class is the main class of the project. It loads all the processor information, loads the template files and clears and populates the attributes to be returned by the ``KordacResult`` object.
+
+- ``Processors/``
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Having folders listed after terms and Python objects is slightly confusing. Possibly define all terms first (define processor there), then have a new section for folders.

Comment (JackMorganNZ) [inline docs/source/contributing.rst]: For the folders section I would also list folders from the top level, so:

- `kordac/processors/`
- `kordac/html-templates/`
- `kordac/tests/`
- `kordac/utils/`
- `kordac/processors-info.json`

and then also:

- `docs`
- `setup.py`
- `requirements.txt`

File: docs/source/contributing.rst [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+- ``processor-info.json``
+	Every processor is listed in this file, and will at least contain a regex pattern to match it's corresponding tag.
+	Most will also define required and optional parameters, these correspond to arguments in the tag's template.
+
+
+
+Creating a New Processor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To create a new processor, a good place to start is the `Extension API`_ page of the Python Markdown docs, or even the `source code`_ itself.
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Suggestion: 

> or even the `source code`_ itself.

to

> and you can also read the `source code`_ itself.

The documentation should be required reading rather than optional.

File: docs/source/contributing.rst [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+.. _Extension API: https://pythonhosted.org/Markdown/extensions/api.html
+
+.. _source code: https://github.com/waylan/Python-Markdown
+
+
+There are several different kinds of processors in Python Markdown, each serving a slightly different purpose. We recommend reading the API docs to determine which processor best suits your purpose (Kordac currently makes use of ``preprocessor``, ``blockprocessor``, ``inlinepattern`` and ``treeprocessor``).
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: We also use `postprocessor`

File: docs/source/contributing.rst [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+The order of the processors matters and is defined when each processor is added to the ``OrderedDict`` in ``KordacExtension.py``.
+
+Each processor is independent of every other processor. If you have two processors in the pipeline that may overlap (e.g. codehilite and fencedcode), the second processor must handle whatever the first outputs, i.e. refrain from manipulating the output of the first processor to help the second.
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Suggestion:

> the second processor must handle whatever the first outputs, i.e. refrain from manipulating the output of the first processor to help the second.

to

> the second processor must handle whatever the first outputs. Therefore refrain from manipulating output of a processor for a later processor.

File: docs/source/contributing.rst [hunk 9]
```diff
@@ -0,0 +1,187 @@
+Contributing to Kordac
+#######################################
+
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Possibly want to provide an introduction about contributing, here is an excerpt from CSFG (some parts may be useful):

> Welcome to the Computer Science Field Guide (CSFG) developer community! We have spent many years creating the CSFG, and we would love for you to get involved into making this guide as great as possible! The following is a set of guidelines for contributing to the CSFG hosted on GitHub. These are just guidelines, not rules, use your best judgment and feel free to propose changes to this document in a pull request.

Probably should also mention Kordac will be/is used for conversion for two large two projects so Kordac needs to remain robust.

File: docs/source/contributing.rst [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+.. _source code: https://github.com/waylan/Python-Markdown
+
+
+There are several different kinds of processors in Python Markdown, each serving a slightly different purpose. We recommend reading the API docs to determine which processor best suits your purpose (Kordac currently makes use of ``preprocessor``, ``blockprocessor``, ``inlinepattern`` and ``treeprocessor``).
+
+The order of the processors matters and is defined when each processor is added to the ``OrderedDict`` in ``KordacExtension.py``.
+
+Each processor is independent of every other processor. If you have two processors in the pipeline that may overlap (e.g. codehilite and fencedcode), the second processor must handle whatever the first outputs, i.e. refrain from manipulating the output of the first processor to help the second.
+
+Every processor belongs in the ``processors/`` directory, but there are several other places they need to be listed, these are:
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Suggestion:

> Every processor belongs in the ``processors/`` directory, but there are several other places they need to be listed, these are:

to

> The logic for each processor belongs in the ``processors/`` directory, and there are several other places processor details need to be listed. These are:

File: docs/source/contributing.rst [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+
+- The frozenset of ``DEFAULT_PROCESSORS`` in ``Kordac.py``
+- The relevant list in ``extendMarkdown()`` in ``KordacExtension.py`` (see `OrderedDict in the Markdown API docs`_ for determining order)
+- The processor's template should be added to ``html-templates`` using the Jinja2 Template Engine syntax for variable parameters
+- The processor's relevant information (regex pattern, required parameters etc) should be included in ``processor-info.json``
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Possibly list this line first as it implies the rest.

File: docs/source/contributing.rst [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Also could note about adding to default list if it should be enabled by default.

File: docs/source/contributing.rst [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+- Be thoroughly tested (see the section below)
+- Have clear and accurate documentation. See the docs on other processors for the preferred format. Your docs should include:
+	- An example of the tag in markdown
+	- Required parameters
+	- Optional parameters
+	- Examples
+	- Examples of overriding the html
+
+We suggest writing the docs before you even write the processor itself as this will give you a clear idea of how a processor in Kordac should behave.
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Suggestion:

> We suggest writing the docs before you

to

> We recommend writing documentation and test cases before you

File: docs/source/contributing.rst [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+
+
+Generally, every processor will have at least three methods (``__init__``, ``test`` and ``run``).
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Probably don't want to list this (or clarify it further) as it's covered in the Markdown extension documentation. 

Also `inlinepatterns` and `postprocessors` don't have `test` method.

File: docs/source/contributing.rst [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+
+The ``test`` method should only return boolean values. This method is used to test if the regex has found a match or not, (and is often used by the Test Suite to check the regex is correct). For Block Processors, this method is also used to determine whether to execute the ``run()`` method or not.
+
+
+The Test Suite
+=======================================
+To start the test suite:
+
+.. code-block:: none
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Set language to `bash` for proper highlighting.

File: docs/source/contributing.rst [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+
+    $ python3 -m kordac.tests.start_tests
+
+This will execute the Smoke, System and then Unit tests.
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Do you want to clarify what each of these test suites are?

File: docs/source/contributing.rst (hunk 1 above)
Comment (hayleyavw) [inline docs/source/contributing.rst]: elaborate?

Comment (JackMorganNZ) [inline docs/source/contributing.rst]: - https://help.github.com/articles/setting-guidelines-for-repository-contributors/
- https://help.github.com/articles/creating-an-issue-template-for-your-repository/
- https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/

We currently use all these for the Computer Science Field Guide.

File: docs/source/contributing.rst [hunk 17]
```diff
@@ -0,0 +1,220 @@
...
+Kordac was created to be used by two much larger projects (the `CS Unplugged`_ and `CS Field Guide`_ websites) as the markdown-to-html converter. The tags we chose are designed to allow authors of these two projects to easily write material without technical elements getting in the way. It is therefore important to us that Kordac remains as simple and robust as possible, please keep this in mind if you decide to work on Kordac with us.
+
+.. _CS Unplugged: https://github.com/uccser/cs-unplugged/
+.. _CS Field Guide: https://github.com/uccser/cs-field-guide/
+
+The git repository for Kordac can be found `here`_, jump in and take a look around!
+
+.. _here: https://github.com/uccser/kordac
+
+<mention django>
```
Comment (ravenmaster001) [inline docs/source/contributing.rst]: Is this a forgotten note.

File: docs/source/contributing.rst [hunk 18]
```diff
@@ -0,0 +1,220 @@
...
+If you come across a bug in Kordac, please `report it on the repo issue tracker`_.
+
+.. _report it on the repo issue tracker: https://github.com/uccser/kordac/issues
+
+If you choose to fix the bug for us, consider adding the relevant tests to the Test Suite (detailed further down this page) to help us catch any future bugs.
+
+
+The Code Base
+=======================================
+If you would like to contribute to Kordac, create a fork of the repository.
```
Comment (ravenmaster001) [inline docs/source/contributing.rst]: Link to github on creating forks.

File: docs/source/contributing.rst [hunk 19]
```diff
@@ -0,0 +1,220 @@
...
+
+Project Structure
+***************************************
+Below is a basic overview of the project structure:
+
+.. code-block:: none
+
+	├── docs/  
+	├── kordac/  
+	│   ├── html-templates/
```
Comment (ravenmaster001) [inline docs/source/contributing.rst]: Was this not renamed to `templates`

File: docs/source/contributing.rst [hunk 20]
```diff
@@ -0,0 +1,220 @@
...
+	│   ├── processor-info.json
+	│   ├── processors/
+	│   │   └── errors/
+	│   ├── tests/
+	│   │   └── assets/
+	│   └── utils/
+	├── requirements.txt
+	└── setup.py
+
+The files and folders of interest are:
```
Comment (ravenmaster001) [inline docs/source/contributing.rst]: Some of these are classes

File: docs/source/contributing.rst [hunk 21]
```diff
@@ -0,0 +1,220 @@
...
+.. _Extension API: https://pythonhosted.org/Markdown/extensions/api.html
+
+.. _source code: https://github.com/waylan/Python-Markdown
+
+
+There are several different kinds of processors in Python Markdown, each serving a slightly different purpose. We recommend reading the API docs to determine which processor best suits your purpose (Kordac currently makes use of ``preprocessor``, ``blockprocessor``, ``inlinepattern``, ``treeprocessor`` and ``postprocessor``).
+
+The order of the processors matters and is defined when each processor is added to the ``OrderedDict`` in ``KordacExtension.py``.
+
+Each processor is independent of every other processor. If you have two processors in the pipeline that may overlap (e.g. codehilite and fencedcode), the second processor must handle whatever the first outputs. Therefore refrain from manipulating the output of a processor for a later processor.
```
Comment (ravenmaster001) [inline docs/source/contributing.rst]: Well they do not have to be independent but they should try to be as much as possible. Other times dependence is necessary as per the ScratchCompatibilityPreprocessor.

If they cannot be independent then compatibility should occur in the processor that happens last (i.e. the downstream processor). That is output should be consistent based on input, not the other way round (e.g. codehilite and fenced_code).

File: docs/source/contributing.rst [hunk 22]
```diff
@@ -0,0 +1,220 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Processor Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+All processor tests inherit from the ``ProcessorTest`` class. Processors should create a ``Mock()`` object, which will contain the bare minimum for the processor to be run (it's jinja template's and properties loaded from ``processor-info.json``), i.e. there is no reason for it to know about properties of the other processors.
+
+A test method will typically follow the same sequence of steps:
+
+	1. Retrieve the test string (there is a ``read_test_file()`` method provided by the ``ProcessorTest`` class)
+	2. Split into blocks (there is a ``to_blocks()`` method provided by the ``ProcessorTest`` class)
```
Comment (ravenmaster001) [inline docs/source/contributing.rst]: Well if it is a blockprocessor yes.

File: docs/source/contributing.rst (hunk 17 above)
Comment (ravenmaster001) [inline docs/source/contributing.rst]: Ignore this, I see it is not finished yet.

File: docs/source/contributing.rst [hunk 23]
```diff
@@ -5,39 +5,28 @@ Welcome to the Kordac developer community! We have spent many months developing
 
 Kordac was created to be used by two much larger projects (the `CS Unplugged`_ and `CS Field Guide`_ websites) as the markdown-to-html converter. The tags we chose are designed to allow authors of these two projects to easily write material without technical elements getting in the way. It is therefore important to us that Kordac remains as simple and robust as possible, please keep this in mind if you decide to work on Kordac with us.
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Suggestion: Use correct capitalisation of languages.

> Markdown to HTML

File: docs/source/contributing.rst [hunk 24]
```diff
@@ -5,39 +5,28 @@ Welcome to the Kordac developer community! We have spent many months developing
...
 
-.. _CS Unplugged: https://github.com/uccser/cs-unplugged/
-.. _CS Field Guide: https://github.com/uccser/cs-field-guide/
-
 The git repository for Kordac can be found `here`_, jump in and take a look around!
 
-.. _here: https://github.com/uccser/kordac
-
-<mention django>
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Did you decide it's not worth discussing Django?

File: docs/source/contributing.rst [hunk 25]
```diff
@@ -170,7 +150,7 @@ There are several arguments that can be used with this command to skip particula
 Test Suite Structure
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Recall our folder structure diagrom from earlier. This time, we're focusing just on the ``tests/`` folder:
+Recall our project structure diagram from earlier. This time, we're focusing just on the ``tests/`` directory:
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Could possibly rephrase this to be:
> We are now focusing on the `tests/` directory in our project structure diagram from earlier:

File: docs/source/contributing.rst [hunk 26]
```diff
@@ -183,57 +163,55 @@ Recall our folder structure diagrom from earlier. This time, we're focusing just
 	        ├── SmokeTests.py
 	        └── start_tests.py
 
-The folder and files of interest are:
+The items of interest are:
 
 - ``BaseTest.py``
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: This uses different notation than `Kordac()` above. Doesn't need to be fixed unless you prefer consistency.

File: docs/source/contributing.rst [hunk 27]
```diff
@@ -183,57 +163,55 @@ Recall our folder structure diagrom from earlier. This time, we're focusing just
...
 - ``SmokeTests.py``
-	Contains smoke tests for both the docs and kordac
+	This contains two classes for smoke testing (one class for ``Kordac``, the other for the docs).
 
 - ``start_tests.py``
+	This is the file that is executed in order to run each of the three types of tests (Smoke, System and Unit). Every new test class must be added to the relevant section of this file.
 
-- - ``assets/``
+- ``assets/``
+	This directory contains a sub directory for every test class that loads external assets (i.e. test input files).
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Possible incorrect use of `i.e.` (I think)

File: docs/source/contributing.rst [hunk 28]
```diff
@@ -183,57 +163,55 @@ Recall our folder structure diagrom from earlier. This time, we're focusing just
...
 
 Adding Tests
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-If you are adding an new test file (e.g. for a new processor), then this needs to be added to the Test Suite in ``start_tests.py``.
-
-
-Provided Base Classes
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When writing a new test function, it is important that the method name is as descriptive as possible, and is prefixed with ``test_``.
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Could add reasoning for prefix.

File: docs/source/contributing.rst [hunk 29]
```diff
@@ -183,57 +163,55 @@ Recall our folder structure diagrom from earlier. This time, we're focusing just
...
 
+If you have added a new processor to ``Kordac``, then a corresponding test suite also needs to be added. This test suite should be added to the ``unit_suite()`` function in ``start_tests.py``. The section below has details on how to write a processor test.
+	
 
 Processor Tests
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-All processor tests inherit from the ``ProcessorTest`` class. Processors should create a ``Mock()`` object, which will contain the bare minimum for the processor to be run (it's jinja template's and properties loaded from ``processor-info.json``), i.e. there is no reason for it to know about properties of the other processors.
+All processor tests inherit from the ``ProcessorTest`` class. Processors should create a ``Mock()`` object, which will contain the bare minimum for the processor to be run (it's html template and properties loaded from ``processor-info.json``), i.e. there is no reason for it to know about properties of the other processors.
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: > html

to

> HTML

File: docs/source/contributing.rst [hunk 30]
```diff
@@ -183,57 +163,55 @@ Recall our folder structure diagrom from earlier. This time, we're focusing just
...
 A test method will typically follow the same sequence of steps:
 
 	1. Retrieve the test string (there is a ``read_test_file()`` method provided by the ``ProcessorTest`` class)
-	2. Split into blocks (there is a ``to_blocks()`` method provided by the ``ProcessorTest`` class)
-	3. Use an ``Assert`` to confirm there are(not) matches to the regex
-	4. Convert the test string using the ``kordac_extension`` (provided by the ``SetUp()`` method in ``ProcessorTest``)
-	5. Load the expected converted result
-	6. Check the converted result is the same as the expected result
-
+	2. Confirm there are(not) matches to the regex in the test string
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Either space needed between `are(not)`, or change to:

> are some/none

File: docs/source/contributing.rst [hunk 31]
```diff
@@ -183,57 +163,55 @@ Recall our folder structure diagrom from earlier. This time, we're focusing just
...
+	3. Convert the test string using the ``kordac_extension`` (provided by the ``SetUp()`` method in ``ProcessorTest``)
+	4. Load the expected converted result
+	5. Check the converted result is the same as the expected result
 
 
 Testing Assets
 ***************************************
 
-<explain markdown and html file >
+Most tests will load an asset file. This file contains example markdown text (and therefore has a ``.md`` extension). For comparing the converted result of this markdown file with it's expected output, a corresponding "expected" file should be created. The expected file should have the same name as the corresponding test file, with ``expected`` appended to the file name.
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: > markdown

to

> Markdown

Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Possibly describe template HTML files too?

File: docs/source/contributing.rst [hunk 32]
```diff
@@ -269,4 +248,20 @@ Kordac should make use GitHub's features:
 Issue template
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Lines still to sort out.

File: docs/source/contributing.rst [hunk 33]
```diff
@@ -269,4 +248,20 @@ Kordac should make use GitHub's features:
...
 Pull request template
 Contributing page
-So GitHub can display these when appropriate.
\ No newline at end of file
+So GitHub can display these when appropriate.
+
+
```
Comment (JackMorganNZ) [inline docs/source/contributing.rst]: Extra whitespace

File: docs/source/contributing.rst (hunk 24 above)
Comment (hayleyavw) [inline docs/source/contributing.rst]: yea well I went through the code and didn't really think it was necessary to talk about it too much.
might add a short note now.

File: docs/source/contributing.rst (hunk 27 above)
Comment (ravenmaster001) [inline docs/source/contributing.rst]: Maybe?
>> This directory consists of further sub-directories containing asset files used in the test suite (i.e. test input files).

File: docs/source/contributing.rst (hunk 26 above)
Comment (hayleyavw) [inline docs/source/contributing.rst]: ah good catch

File: .github/ISSUE_TEMPLATE.md [hunk 34]
```diff
@@ -0,0 +1,21 @@
...
+
+
+
+
+
+
+
+### Checklist
+
+*Put an `x` in the boxes that apply. You can also fill these out after creating the issue.*
```
Comment (JackMorganNZ) [inline .github/ISSUE_TEMPLATE.md]: Probably want to say 'Change the space in the box to an 'x' for those that apply', as I constantly get people doing `[x ]` on the CSFG.

File: CONTRIBUTING.md [hunk 35]
```diff
@@ -0,0 +1,59 @@
...
+
+We've written a technical guide about how Kordac works and how to contribute [here](kordac.readthedocs.io/en/develop/contributing.html). These docs should give you a good overview of how Kordac is pieced together.
+
+Below are a few more general notes to remember while you are working on Kordac.
+
+### Code of Conduct
+
+This project adheres to the Contributor Covenant [code of conduct](CODE_OF_CONDUCT.md).
+By participating, you are expected to uphold this code.
+Please report unacceptable behavior to [csse-education-research@canterbury.ac.nz](csse-education-research@canterbury.ac.nz)
```
Comment (JackMorganNZ) [inline CONTRIBUTING.md]: Missing `mailto:`

**Comment by hayleyavw** [comment]
ping @ravenmaster001 If you're happy with this then I'll merge it in :)

**Comment by ravenmaster001** [comment]
@hayleyavw was just reading it this morning, sorry it took so long

**Comment by ravenmaster001** [comment]
Uhhhh.... I pressed the wrong button...
